tests/unit/core/utils/test_compatibility.js

```diff
@@ -319,105 +311,100 @@ describe(__filename, () => {
+        os: { name: 'Windows' },
+      };
+      expect(
+        _isCompatibleWithUserAgent({
+          minVersion: '10.1',
+          userAgentInfo,
+        }),
+      ).toEqual({ compatible: false, reason: INCOMPATIBLE_UNDER_MIN_VERSION });
     });
   });
 
   describe('getCompatibleVersions', () => {
+    const _getCompatibleVersions = ({
+      addon = createInternalAddon(fakeAddon),
+      currentVersion = createInternalVersion(fakeVersion),
+      ...rest
+    }) => {
+      return getCompatibleVersions({
+        addon,
+        currentVersion,
+        ...rest,
+      });
+    };
+
     it('gets the min and max versions', () => {
-      const addon = createInternalAddon({
-        ...fakeAddon,
-        current_version: {
-          ...fakeAddon.current_version,
-          compatibility: {
-            firefox: {
-              max: '20.0.*',
-              min: '11.0.1',
-            },
+      const currentVersion = createInternalVersion({
+        ...fakeVersion,
+        compatibility: {
+          firefox: {
+            max: '20.0.*',
+            min: '11.0.1',
           },
         },
       });
-      const { maxVersion, minVersion } = getCompatibleVersions({
-        addon,
+
+      const { maxVersion, minVersion } = _getCompatibleVersions({
         clientApp: CLIENT_APP_FIREFOX,
+        currentVersion,
       });
 
       expect(maxVersion).toEqual('20.0.*');
       expect(minVersion).toEqual('11.0.1');
     });
 
     it('gets null if the clientApp does not match', () => {
-      const addon = createInternalAddon({
-        ...fakeAddon,
-        current_version: {
-          ...fakeAddon.current_version,
-          compatibility: {
-            firefox: {
-              max: '20.0.*',
-              min: '11.0.1',
-            },
+      const currentVersion = createInternalVersion({
+        ...fakeVersion,
+        compatibility: {
+          firefox: {
+            max: '20.0.*',
+            min: '11.0.1',
           },
         },
       });
-      const { maxVersion, minVersion } = getCompatibleVersions({
-        addon,
+
+      const { maxVersion, minVersion } = _getCompatibleVersions({
         clientApp: CLIENT_APP_ANDROID,
+        currentVersion,
       });
 
       expect(maxVersion).toEqual(null);
       expect(minVersion).toEqual(null);
     });
 
     it('returns null if clientApp has no compatibility', () => {
-      const addon = createInternalAddon({
-        ...fakeAddon,
-        current_version: {
-          ...fakeAddon.current_version,
-          compatibility: {},
-        },
-      });
-      const { maxVersion, minVersion } = getCompatibleVersions({
-        addon,
-        clientApp: CLIENT_APP_FIREFOX,
+      const currentVersion = createInternalVersion({
+        ...fakeVersion,
+        compatibility: {},
       });
 
-      expect(maxVersion).toEqual(null);
-      expect(minVersion).toEqual(null);
-    });
-
-    it('returns null if current_version does not exist', () => {
```

Comment:
What happened to this feature? Is it still supported in some way? We need to support add-ons without `current_version` for the case of incomplete add-ons. You can view this type of add-on via AMO (not DiscoPane) as an admin and sometimes as a developer.

Reply:
`getCompatibleVersions` no longer accesses `current_version` but rather accepts `currentVersion` as an argument, and it is required to have a value. So this test is no longer meaningful. I've added an `invariant` into `getCompatibleVersions` to enforce it in the case that Flow doesn't catch it.

Reply:
> `currentVersion` ... is required to have a value. I've added an invariant into `getCompatibleVersions` to enforce it in the case that Flow doesn't catch it.

What I'm saying is that a `null` `currentVersion` object is something we need to support in real life. You can get an add-on with a `null` `current_version` property by looking at the detail page for an incomplete submission either as an admin or a developer.

Reply:
Ah, I see the issue. I am dealing with that currently by *not calling* `getClientCompatibility` from `Addon` if `currentVersion` is falsey. I do the check in `Addon` as that's the only place where this is called. Flow ensures that we don't try to call `getClientCompatibility` without a `currentVersion`, but I guess my current approach is less encapsulated.  I really should add back in the logic to accept a `null` for `currentVersion` and just return a null in that case. I'll do that.